### PR TITLE
openal: 1.22.0 -> 1.22.2

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/openal-soft/default.nix [hunk 1]
```diff
@@ -25,6 +25,13 @@ stdenv.mkDerivation rec {
       --replace "@OUT@" $out
   '';
 
+  postInstall = ''
+    # the pc file contains the nix store path twice
+    substituteInPlace $out/lib/pkgconfig/openal.pc \
+      --replace "libdir=\''${exec_prefix}/$out/lib" "libdir=\''${exec_prefix}/lib" \
+      --replace "includedir=\''${prefix}/$out/include" "includedir=\''${prefix}/include"
```
Comment (bjornfor) [inline pkgs/development/libraries/openal-soft/default.nix:32]: I looked upstream and don't understand why we end up with this broken file:

* https://github.com/kcat/openal-soft/blob/master/openal.pc.in
* https://github.com/kcat/openal-soft/blob/master/CMakeLists.txt#L1262-L1266

Comment (bjornfor) [reply pkgs/development/libraries/openal-soft/default.nix:32]: Ah, we need to bump the version, the pkg-config file was fixed upstream in v1.2.21. Latest version is v1.2.22.

**Review by bjornfor — CHANGES_REQUESTED** [review]
Please bump the version instead as this issue has been fixed upstream already. (Was fixed in 1.22.1, latest version is 1.22.2).

File: pkgs/development/libraries/openal-soft/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/development/libraries/openal-soft/default.nix]: done.

**Review by bjornfor — APPROVED** [review]

**Pull request merged**